testsuite/MDAnalysisTests/coordinates/test_rdkit.py

```diff
@@ -20,59 +20,480 @@
+
+
+@pytest.fixture(scope="function")
+def product(request):
+    return getattr(MolFactory, request.param)()
+
+
+@requires_rdkit
 class TestRDKitReader(object):
     @pytest.mark.parametrize("rdmol, n_frames", [
-        (mol2_mol(), 1),
-        (smiles_mol(), 3),
-    ])
+        ("mol2_mol", 1),
+        ("smiles_mol", 3),
+    ], indirect=["rdmol"])
     def test_coordinates(self, rdmol, n_frames):
         universe = mda.Universe(rdmol)
         assert universe.trajectory.n_frames == n_frames
         expected = np.array([
-            conf.GetPositions() for conf in rdmol.GetConformers()], 
+            conf.GetPositions() for conf in rdmol.GetConformers()],
             dtype=np.float32)
         assert_equal(expected, universe.trajectory.coordinate_array)
 
     def test_no_coordinates(self):
-        with warnings.catch_warnings(record=True) as w:
-            # Cause all warnings to always be triggered.
-            warnings.simplefilter("always")
-            # Trigger a warning.
+        with pytest.warns(UserWarning, match="No coordinates found"):
             u = mda.Universe.from_smiles("CCO", generate_coordinates=False)
-            # Verify the warning
-            assert len(w) == 1
-            assert "No coordinates found" in str(
-                w[-1].message)
-        expected = np.empty((1,u.atoms.n_atoms,3), dtype=np.float32)
+        expected = np.empty((1, u.atoms.n_atoms, 3), dtype=np.float32)
         expected[:] = np.nan
         assert_equal(u.trajectory.coordinate_array, expected)
 
     def test_compare_mol2reader(self):
-        universe = mda.Universe(mol2_mol())
+        universe = mda.Universe(MolFactory.mol2_mol())
         mol2 = mda.Universe(mol2_molecule)
         assert universe.trajectory.n_frames == mol2.trajectory.n_frames
-        assert_equal(universe.trajectory.ts.positions, 
+        assert_equal(universe.trajectory.ts.positions,
                      mol2.trajectory.ts.positions)
-        
+
+
+@requires_rdkit
+class TestRDKitConverter(object):
+    @pytest.fixture
+    def pdb(self):
+        return mda.Universe(PDB_full)
+
+    @pytest.fixture
+    def mol2(self):
+        u = mda.Universe(mol2_molecule)
+        # add elements
+        elements = np.array([guess_atom_element(x) for x in u.atoms.types],
+                            dtype=object)
+        u.add_TopologyAttr('elements', elements)
+        return u
+
+    @pytest.fixture
+    def peptide(self):
+        u = mda.Universe(GRO)
+        elements = mda.topology.guessers.guess_types(u.atoms.names)
+        u.add_TopologyAttr('elements', elements)
+        return u.select_atoms("resid 2-12")
+
+    @pytest.mark.parametrize("smi", ["[H]", "C", "O", "[He]"])
+    def test_single_atom_mol(self, smi):
+        u = mda.Universe.from_smiles(smi, addHs=False,
+                                     generate_coordinates=False)
+        mol = u.atoms.convert_to("RDKIT")
+        assert mol.GetNumAtoms() == 1
```

Comment:
This might be a simple enough case for testing that the right elements were assigned to the mol?